File: src/main/java/com/xebia/xboson/IdSelectionRule.java
package com.xebia.xboson;

import java.util.Optional;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

public class IdSelectionRule implements SelectionRule {
    @Override
    public Optional<ParsedElement> match(Document document, Element element) {
        String id = element.id();
        if ("".equals(id)) {
            return Optional.empty();
        }
        String cssQuery = String.format("#%s", id);
        Elements found = document.select(cssQuery);
        if (found.size() == 1) {
            return Optional.of(new ParsedElement(id, cssQuery));
        }else {
            return Optional.empty();
        }
    }
}

File: src/main/kotlin/com/xebia/xboson/IdSelectionRule.kt
package com.xebia.xboson

import java.util.Optional
import org.jsoup.nodes.Document
import org.jsoup.nodes.Element
import org.jsoup.select.Elements

class IdSelectionRule : SelectionRule {
    override fun match(document: Document, element: Element): ParsedElement? {
        val id = element.id()
        if ("" == id) {
            return null
        }
        val cssQuery = String.format("#%s", id)
        val found = document.select(cssQuery)
        if (found.size == 1) {
            return ParsedElement(id, cssQuery)
        } else {
            return null
        }
    }
}
